src/backend/access/external/pxfutils.c

```diff
@@ -39,7 +39,7 @@ bool are_ips_equal(char *ip1, char *ip2)
 /* override port str with given new port int */
 void port_to_str(char **port, int new_port)
 {
-	char tmp[10];
+	char tmp[sizeof(new_port) + 2];
```

Comment:
This will not be sufficient. sizeof would return 4 and tmp would be allocated only 6 bytes which wouldn't be sufficient to store the 32 bit interger as a string